services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::time::SystemTime;
+
+use splinter::error::InvalidStateError;
+use splinter::service::FullyQualifiedServiceId;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SupervisorNotification {
+    service_id: FullyQualifiedServiceId,
+    action_id: i64,
+    notification_type: SupervisorNotificationType,
+    request_for_vote_value: Option<Vec<u8>>,
+    created_at: Option<SystemTime>,
+    executed_at: Option<SystemTime>,
+}
+
+impl SupervisorNotification {
+    /// Returns the service ID for the notficiation
+    pub fn service_id(&self) -> &FullyQualifiedServiceId {
+        &self.service_id
+    }
+
+    /// Returns the id of the associated action
+    pub fn action_id(&self) -> &i64 {
+        &self.action_id
+    }
+
+    /// Returns the notification_type for the notficiation
+    pub fn notification_type(&self) -> &SupervisorNotificationType {
+        &self.notification_type
+    }
+
+    /// Returns the value for the notficiation
+    pub fn request_for_vote_value(&self) -> &Option<Vec<u8>> {
```

Comment:
Consider include this in the enum to avoid the Option

Reply:
Updated to remove request_for_vote_value and use `SupervisorNotificationType::ParticipantRequestForVote { value: Vec<u8> },` instead